### PR TITLE
feat(manage): improve breadcrumbs usability

### PR DESCRIPTION
**Pull request by wescopeland**
This PR slightly alerts the management app's achievement and leaderboard page breadcrumbs so the game is always in the list of links.

**Before**
Screenshot 2024-08-05 at 7 05 46 PM

**After**
Screenshot 2024-08-05 at 7 07 20 PM

### CODE REVIEW

File: app/Filament/Resources/LeaderboardResource/Pages/Details.php [hunk 1]
```diff
@@ -5,13 +5,28 @@
...
+    {
+        /** @var Leaderboard $leaderboard */
+        $leaderboard = $this->record;
+        $game = $leaderboard->game;
+
+        return [
+            route('filament.admin.resources.leaderboards.index') => 'Leaderboards',
+            route('filament.admin.resources.games.view', $game) => $game->title,
+            route('filament.admin.resources.leaderboards.view', $leaderboard) => $leaderboard->title,
+            'Audit Log',
```
Comment (Jamiras) [inline app/Filament/Resources/LeaderboardResource/Pages/Details.php:26]: This is not the audit log

File: app/Filament/Resources/LeaderboardResource/Pages/Edit.php [hunk 2]
```diff
@@ -5,13 +5,28 @@
...
+    {
+        /** @var Leaderboard $leaderboard */
+        $leaderboard = $this->record;
+        $game = $leaderboard->game;
+
+        return [
+            route('filament.admin.resources.leaderboards.index') => 'Leaderboards',
+            route('filament.admin.resources.games.view', $game) => $game->title,
+            route('filament.admin.resources.leaderboards.view', $leaderboard) => $leaderboard->title,
+            'Audit Log',
```
Comment (Jamiras) [inline app/Filament/Resources/LeaderboardResource/Pages/Edit.php:26]: This is not the audit log

**Review by Jamiras — COMMENTED** [review]
Somewhat related: the leaderboard link on the game page goes directly to the edit sub-page.
image

**Comment by wescopeland** [comment]
> Somewhat related: the leaderboard link on the game page goes directly to the edit sub-page.

This is a default Filament behavior for users who have the ability to edit the resource. If the user doesn't have the ability to edit the resource, they'll instead be linked to the resource's "Details" (view) page.

With that in mind, we should probably ensure both pages follow a similar layout & show the same fields. I'll make a note to circle back on this in a follow-up.

File: app/Filament/Resources/LeaderboardResource/Pages/Details.php (hunk 1 above)
Comment (wescopeland) [reply app/Filament/Resources/LeaderboardResource/Pages/Details.php]: Thanks, fixed in latest.

File: app/Filament/Resources/LeaderboardResource/Pages/Edit.php (hunk 2 above)
Comment (wescopeland) [reply app/Filament/Resources/LeaderboardResource/Pages/Edit.php]: Thanks, fixed in latest.

**Comment by Jamiras** [comment]
> This is a default Filament behavior for users who have the ability to edit the resource. If the user doesn't have the ability to edit the resource, they'll instead be linked to the resource's "Details" (view) page.

Odd. The achievement links weren't directed to the edit page for the achievement, and I had permission to edit those too.

**Comment by wescopeland** [comment]
Ah, that is because in _AchievementsRelationManager.php_, for some reason I put in an override:
```php
->recordUrl(
    fn (Achievement $record): string => route('filament.admin.resources.achievements.view', ['record' => $record])
)
```

Now that you mention it, this should probably be removed as it creates an inconsistent user experience.

**Comment by wescopeland** [comment]
Although, with that override removed, the table rows don't link anywhere at all. That must explain why I added it.

**Comment by wescopeland** [comment]
I've made a change to those achievement rows in https://github.com/RetroAchievements/RAWeb/pull/2595/commits/2c80aa48c80364f50062ccc92ae34e62ff136daa so we can have a consistent behavior. The rows now do a policy check to determine what the href should be.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**